### PR TITLE
Contract changes draft

### PR DESCRIPTION
**Pull request by vivianyyd**

### CODE REVIEW

File: runtime/src/main/kotlin/io/github/apl_cornell/viaduct/runtime/Boxed.kt [hunk 1]
```diff
@@ -1,13 +1,13 @@
 package io.github.apl_cornell.viaduct.runtime
 
-abstract class Boxed {
-    abstract var temporary: Any
+/** Represents a mutable cell. */
+class Boxed<T>(var value: T) {
```
Comment (cacay) [inline runtime/src/main/kotlin/io/github/apl_cornell/viaduct/runtime/Boxed.kt:4]: Mark this as private.

File: compiler/src/main/kotlin/io/github/apl_cornell/viaduct/codegeneration/BackendCodeGenerator.kt [hunk 2]
```diff
@@ -229,6 +238,60 @@
...
+    private fun let(protocol: Protocol, stmt: LetNode): CodeBlock =
+        CodeBlock.of(
+            "val %N = %L",
+            context.kotlinName(stmt.name.value, protocol),
+            codeGenerator.exp(protocol, stmt.value)
+        )
+
+    private fun declaration(protocol: Protocol, stmt: DeclarationNode): CodeBlock {
+        val ctorCall = codeGenerator.constructorCall(protocol, stmt.objectType, stmt.arguments)
+        return when (stmt.objectType.className.value) {
```
Comment (cacay) [inline compiler/src/main/kotlin/io/github/apl_cornell/viaduct/codegeneration/BackendCodeGenerator.kt:268]: No need to pattern match here. Every case is the same.

File: compiler/src/main/kotlin/io/github/apl_cornell/viaduct/codegeneration/BackendCodeGenerator.kt [hunk 3]
```diff
@@ -229,6 +238,60 @@
...
+
+            is OutputNode -> CodeBlock.of(
+                "runtime.output(%T(%L))",
+                typeAnalysis.type(stmt.message).valueClass,
+                codeGenerator.exp(protocol, stmt.message)
+            )
+        }
+    }
+
+    private fun let(protocol: Protocol, stmt: LetNode): CodeBlock =
```
Comment (cacay) [inline compiler/src/main/kotlin/io/github/apl_cornell/viaduct/codegeneration/BackendCodeGenerator.kt:259]: These (`let`, `declaration`, `outParameterInitialization` etc.) are short enough that you should just inline them.

**Review by cacay — CHANGES_REQUESTED** [review]
Looks good! I recommended some small changes; we can merge after.

**Comment by cacay** [comment]
This PR looks complete. Should it be marked as not draft?

File: runtime/src/main/kotlin/io/github/apl_cornell/viaduct/runtime/Boxed.kt (hunk 1 above)
Comment (vivianyyd) [reply runtime/src/main/kotlin/io/github/apl_cornell/viaduct/runtime/Boxed.kt:4]: Philisophical question: What's the point of having a private variable with set() and get() to begin with? 
Forgot to bring this up during our meeting but I'm having trouble understanding what's wrong with just exposing the value

Comment (vivianyyd) [reply runtime/src/main/kotlin/io/github/apl_cornell/viaduct/runtime/Boxed.kt:4]: Forgot to bring this up during our meeting, but why use a private variable with get() set() at all? If the only aim is to box up a value, the public variable alone seems enough

Comment (cacay) [reply runtime/src/main/kotlin/io/github/apl_cornell/viaduct/runtime/Boxed.kt]: Doing `.value` is annoying? There isn't a great reason to use `get()` and `set()` I guess, but it's fine.

**Pull request merged**